.github/workflows/setup-terraform.yml

```diff
@@ -0,0 +1,310 @@
+name: 'Setup Terraform'
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  opentofu-versions:
+    name: 'Terraform Versions'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        opentofu-versions: [0.11.14, latest]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
+
+    - name: Setup Terraform - ${{ matrix['opentofu-versions'] }}
+      uses: ./
+      with:
+        opentofu_version: ${{ matrix['opentofu-versions'] }}
+
+    - name: Validate Teraform Version - ${{ matrix['opentofu-versions'] }}
+      if: ${{ matrix['opentofu-versions'] != 'latest' }}
+      run: opentofu version | grep ${{ matrix['opentofu-versions']}}
+
+    - name: Validate Teraform Version - ${{ matrix['opentofu-versions'] }}
+      if: ${{ matrix['opentofu-versions'] == 'latest' }}
+      run: opentofu version | grep 'Terraform v'
+
+  opentofu-versions-no-wrapper:
+    name: 'Terraform Versions No Wrapper'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        opentofu-versions: [0.11.14, latest]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
+
+    - name: Setup Terraform (no wrapper) - ${{ matrix['opentofu-versions'] }}
+      uses: ./
+      with:
+        opentofu_version: ${{ matrix['opentofu-versions'] }}
+        opentofu_wrapper: false
+
+    - name: Validate Teraform Version - ${{ matrix['opentofu-versions'] }}
+      if: ${{ matrix['opentofu-versions'] != 'latest' }}
+      run: opentofu version | grep ${{ matrix['opentofu-versions']}}
+
+    - name: Validate Teraform Version - ${{ matrix['opentofu-versions'] }}
+      if: ${{ matrix['opentofu-versions'] == 'latest' }}
+      run: opentofu version | grep 'Terraform v'
+
+  opentofu-versions-constraints:
+    name: 'Terraform Versions Constraints'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        opentofu-versions: [~0.12, 0.12.x, <0.13.0]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
+
+    - name: Setup Terraform - ${{ matrix['opentofu-versions'] }}
+      uses: ./
+      with:
+        opentofu_version: ${{ matrix['opentofu-versions'] }}
+
+    - name: Validate Teraform Version - ${{ matrix['opentofu-versions'] }}
+      run: opentofu version | grep 'Terraform v0\.12'
```

Comment:
@imjasonh You mentioned that the tests have passed - it's an indication that unfortunately this codebase does only rename from terraform to opentofu without pulling the actual `tofu` binary. Please compare to the tests definition in #2. Thanks!